File: 2023-2024/sourcecode/mod1/lab4/32TabLayoutViewPager2NavigationBasicsJava/app/src/main/java/com/example/a32_tablayoutviewpager2navigationbasics_java/PagerAdapter.java
package com.example.a32_tablayoutviewpager2navigationbasics_java;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class PagerAdapter extends FragmentStateAdapter {

    public static final String arg = "key";

    public PagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment fragment = new TemplateFragment();
        Bundle args = new Bundle();
        args.putInt(arg, position + 1);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: 2023-2024/sourcecode/mod1/lab4/31TabLayoutViewPager2NavigationBasicsKotlin/app/src/main/java/com/example/a31_tablayoutviewpager2navigationbasics_kotlin/PagerAdapter.kt
package com.example.a31_tablayoutviewpager2navigationbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

const val arg ="key"

class PagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
    override fun getItemCount(): Int = 4

    override fun createFragment(position: Int): Fragment {
        val fragment = TemplateFragment()
        fragment.arguments = Bundle().apply {
            putInt(arg, position + 1)
        }

        return fragment
    }
}